### ISSUE TITLE
[Oracle] Purifying Flames puts ally's Blink Dagger on 3 seconds cooldown

### ISSUE THREAD

**Issue by buzlad** [ValveSoftware/Dota2-Gameplay#7818]
### Ability name

Purifying Flames

### Description

Oracle's Purifying Flames cancels his teammate's Blink Dagger, Healing Salve, Clarity and Bottle despite their tooltips indicating otherwise.
- Blink Dagger: Blink Dagger cannot be used for 3 seconds after taking damage from an enemy hero or Roshan.
- Healing Salve: If the unit is attacked by an enemy hero or Roshan, the effect is lost.

Please note that the items will not be cancel if
- target takes no damage from Purifying Flames, for example using Purifying Flames when the target is under effect of Fate's Edict or False Promise.
- Oracle is targeting himself, as it is self-damage just like Pudge's Rot.

Edit: just noticed related issue #5235 

### Example Match ID (and possibly Timestamp)

7087870783 - 47:00, replicated in demo mode

**Comment by Mullins22W** [comment]
That is correct. The Purifying Flames ability of Oracle in the game Dota 2 can be used on allies to deal damage to enemies or heal allies. When used on an ally, it also puts their Blink Dagger on a 3-second cooldown.

**Comment by TheZett** [comment]
This is a tooltip issue on Blink Dagger and other items that react on hero-based on-damage events.

Any hero-based damage, except for self-inflicted damage, causes Blink Dagger to go onto its 3s damage cooldown.
Functionally this is intended. Purifying Flames also procs allied Raindrops.

**Comment by buzlad** [comment]
> This is a tooltip issue on Blink Dagger and other items that react on hero-based on-damage events.
> 
> Any hero-based damage, except for self-inflicted damage, causes Blink Dagger to go onto its 3s damage cooldown. Functionally this is intended. Purifying Flames also procs allied Raindrops.

**Comment by buzlad** [comment]
> This is a tooltip issue on Blink Dagger and other items that react on hero-based on-damage events.
> 
> Any hero-based damage, except for self-inflicted damage, causes Blink Dagger to go onto its 3s damage cooldown. Functionally this is intended. Purifying Flames also procs allied Raindrops.

I understand that if the tooltips are updated accordingly, then it should be the correct interaction. However I would lean more into that it's a bug on the ability itself.

I also just tested this in demo mode, Purifying Flames does not proc on allied Raindrops.
